test/y2storage/encryption_method_test.rb

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env rspec
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require "y2storage/encryption_method"
+
+describe Y2Storage::EncryptionMethod do
+  describe ".all" do
+    it "returns a method for Luks1" do
+      expect(described_class.all.map(&:to_sym)).to include(:luks1)
+    end
+
+    it "returns a method for a random swap" do
+      expect(described_class.all.map(&:to_sym)).to include(:random_swap)
+    end
+  end
+
+  describe ".find" do
+    context "when looking for a known method" do
+      it "returns the encryption method" do
+        luks1 = described_class.find(:luks1)
+        random_swap = described_class.find(:random_swap)
+
+        expect(luks1).to be_a Y2Storage::EncryptionMethod
+        expect(luks1.id).to eq(:luks1)
+
+        expect(random_swap).to be_a Y2Storage::EncryptionMethod
+        expect(random_swap.id).to eq(:random_swap)
+      end
+    end
+
+    context "when looking for an unknown method" do
+      it "returns nil" do
+        expect(described_class.find("unknown value")).to be_nil
+      end
+    end
+  end
+
+  describe "#create_device" do
```

Comment:
`luks1` and `random_swap` encryption methods have quite different behavior when creating a device. I would add tests for each one, checking for example the type of encryption it creates (plain or luks).